code/game/machinery/camera/camera.dm

```diff
@@ -47,13 +47,17 @@
 	assembly.state = 4
 	GLOB.cameranet.cameras += src
 	GLOB.cameranet.addCamera(src)
+	var/area/A = get_area(src)
+	LAZYADD(A.cameras, src)
 	proximity_monitor = new(src, 1)
 
 	if(mapload && (z in GLOB.station_z_levels) && prob(3) && !start_active)
 		toggle_cam()
 
 /obj/machinery/camera/Destroy()
 	toggle_cam(null, 0) //kick anyone viewing out
+	var/area/A = get_area(src)
+	LAZYREMOVE(A.cameras, src)
```

Comment:
you need to account for the possibility of cameras moving, due to stuff like the singularity.

Reply:
Can cameras really be moved without destroying them? I thought about it while making this but I guess I'll have to test it out a bit

Reply:
Dont borgs contain a camera?

Reply:
All silicons potentially do. Those are fixed.